objects/client.c

```diff
@@ -371,14 +371,33 @@
  */
 
 /**
- * The client icon.
+ * The client icon as a surface (mapped memory).
+ *
+ * This property holds the client icon closest to the size configured via
+ * @{awesome.set_preferred_icon_size}.
+ *
+ * It is a bitmap mapped in memory and not a file path nor a widget.
+ *
+ * Typically you would want to use @{awful.widget.clienticon} to get this as a
+ * widget.
+ *
+ * If you want to copy it into a cairo surface you should do something along,
+ * since it would otherwise crash awesome in case it gets accessed after the
+ * client is removed.
```

Comment:
Nope. You can just do `local s = gears.surface(c.icon)` and are good.
However, this `gears.surface`-call (must be called on the returned value to avoid memory leaks, must not be called more than once to avoid double-frees (= crashes)) is quite important and I'd rather add this more prominently here.

Reply:
Umm, yes, I remembered there was something to say, but that was a very long time ago when you explained that one to me. I forgot that peculiarity and instead used the copy to avoid getting the issue.

I am not sure however how relevant saying it must not be done twice is for an end user. There is no way to controls this for the user. If he/she/it uses a tasklist and an 3rd party alt-tab menu that happens to also create a surface, then the copy seems a better idea than a dead-end.

In the end both option suck and maybe the real one is to implement a `awful.client.object.get_icon` and rename the "real" `c.icon` to `c._icon` and handle this in Lua? The average user should not be able to double-free Awesome accidentally.

For now I will add both in the doc, but I consider this a bug.